nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/AbstractTrashService.java

```diff
@@ -112,19 +117,60 @@ public boolean canPurgeOrUntrash(List<DocumentModel> docs, NuxeoPrincipal princi
             return false;
         }
         // used to do only check on parent perm
-        TrashInfo info = getInfo(docs, principal, false, true);
+        TrashInfo info = getInfo(docs, principal, false, true, true);
         return info.docs.size() == docs.size();
     }
 
     protected TrashInfo getInfo(List<DocumentModel> docs, NuxeoPrincipal principal, boolean checkProxies,
             boolean checkDeleted) {
+        return getInfo(docs, principal, checkProxies, checkDeleted, false);
+    }
+
+    @Override
+    public boolean getCanEmptyTrash(DocumentModel parent, NuxeoPrincipal principal) {
+        String nxql = String.format(TRASHED_QUERY, parent.getId());
+        ScrollRequest request = DocumentScrollRequest.builder(nxql)
+                                                     .username(principal.getName())
+                                                     .repository(parent.getRepositoryName())
+                                                     .build();
+        ScrollService service = Framework.getService(ScrollService.class);
+        boolean canPurge = true;
+        try (Scroll scroll = service.scroll(request)) {
+            if (!scroll.hasNext()) {
+                return false;
+            }
+            while (scroll.hasNext() && canPurge) {
+                DocumentModelList documents = RepositoryScroll.loadDocuments(parent.getCoreSession(), scroll.next());
+                canPurge = canPurgeOrUntrash(documents, principal);
```

Comment:
here you don't to call `RespositoryScroll#loadDocuments`  you don't scroll a second time ? is it right 

just make query on the `CoreSession#query` or extract a method in the `TrashService` but we should care about what `scroll.next()` returns i means how the size of the documents list will be, because in some SQL DB there is some limitation when you make `IN` for example Oracle don't accepts more then `1000` with `IN` clause

Reply:
default scroll size is 50